ui/src/components/ui/device/device-screen/device-screen.tsx

```diff
@@ -0,0 +1,213 @@
+import { useRef } from 'react'
+import cn from 'classnames'
+import { useParams } from 'react-router-dom'
+import { observer } from 'mobx-react-lite'
+import { Spinner } from '@vkontakte/vkui'
+
+import { ConditionalRender } from '@/components/lib/conditional-render'
+
+import { touchService } from '@/services/touch-service/touch-service'
+import { keyboardService } from '@/services/keyboard-service/keyboard-service'
+
+import { deviceScreenStore } from '@/store/device-screen-store/device-screen-store'
+import { useScreenAutoQuality } from '@/lib/hooks/use-screen-auto-quality.hook'
+import { useScreenStreaming } from '@/lib/hooks/use-screen-streaming.hook'
+import { deviceBySerialStore } from '@/store/device-by-serial-store'
+
+import styles from './device-screen.module.css'
+
+import type { ChangeEvent, ClipboardEvent, KeyboardEvent, MouseEvent, TouchEvent } from 'react'
+
+export const DeviceScreen = observer(() => {
+  const canvasWrapperRef = useRef<HTMLDivElement>(null)
+  const canvasRef = useRef<HTMLCanvasElement>(null)
+  const inputRef = useRef<HTMLInputElement>(null)
+  const { serial } = useParams()
+
+  const { data: device } = deviceBySerialStore.deviceQueryResult(serial || '')
+
+  useScreenStreaming({ canvasRef, canvasWrapperRef, serial })
+  useScreenAutoQuality(serial)
+
+  const onMouseDown = (event: MouseEvent<HTMLDivElement>) => {
+    if (!serial) return
+
+    event.preventDefault()
+
+    touchService.mouseDownListener({
```

Comment:
At this point this separation of ui/business logic doesn't make sense. I think it's better for touchService to be aware of the ui and handle the whole DOM Event: i.e. move the preventDefault and serial there. But I can't stand up for it - i'm not sure